lib/usd/translators/nurbsCurveWriter.cpp

```diff
@@ -159,18 +200,26 @@ bool PxrUsdTranslators_NurbsCurveWriter::writeNurbsCurveAttrs(
     MDoubleArray mayaCurveKnots;
     status = curveFn.getKnots(mayaCurveKnots);
     CHECK_MSTATUS_AND_RETURN(status, false);
-    VtDoubleArray curveKnots(mayaCurveKnots.length() + 2); // all knots batched together
-    for (unsigned int i = 0; i < mayaCurveKnots.length(); i++) {
-        curveKnots[i + 1] = mayaCurveKnots[i];
-    }
+    const uint32_t mayaKnotsCount = mayaCurveKnots.length();
+    VtDoubleArray  curveKnots;
+    auto           copyKnots = [&mayaCurveKnots, &curveKnots](size_t from, size_t to) {
+        memcpy(
+            ((double*)curveKnots.cdata()) + from,
+            (const double*)&mayaCurveKnots[0],
+            sizeof(double) * to);
+    };
     if (wrap) {
+        // Insert wrapping knots at either end of the vector
+        curveKnots.resize(mayaKnotsCount + 2);
+        copyKnots(1, mayaKnotsCount + 1);
```

Comment:
I don't understand why this call to copyKnots needs to copy mayaKnotsCount + 1 knots. Doesn't that read beyond the end of the data in mayaCurveKnots?

The code does immediately overwrite whatever got copied into curveKnots, but reading that extra data is still incorrect.

If my interpretation is correct the copyKnots lambda doesn't need the to parameter, it could capture mayaKnotsCount because it should always copy the same amount of data.

Reply:
Sorry @williamkrick you are quite correct, must have mistaken that third param for an index. I'll fix that right up.